internal/actions/create_handler.go

```diff
@@ -53,21 +50,26 @@ func (h *CreateHandler) Handle(ctx context.Context, action *castai.ClusterAction
 		"name":           newObj.GetName(),
 	})
 
-	r := h.client.Resource(schema.GroupVersionResource{
+	gvkResource := h.client.Resource(schema.GroupVersionResource{
 		Group:    req.Group,
 		Version:  req.Version,
 		Resource: req.Resource,
-	}).Namespace(newObj.GetNamespace())
+	})
+
+	var resource dynamic.ResourceInterface = gvkResource
+	if newObj.GetNamespace() != "" {
```

Comment:
Can you please add unit tests to cover this?

Reply:
done